Feat/teaching materials from owa trpc

JBR90 opened the pull request:
## Description

- List of changes

## Issue(s)

Fixes #

## How to test

1. Go to {deployment_url}
2. Click on \_\_\_\_\_\_
3. You should see \_\_\_\_\_\_

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Does this PR update a package with a breaking change

JBR90 commented on apps/nextjs/src/app/aila/teaching-materials/teachingMaterialsView.tsx:
```diff
@@ -120,20 +109,27 @@ const TeachingMaterialsViewInner: FC<AdditionalMaterialsUserProps> = () => {
-    >
-      {stepComponents[stepNumber]}
-    </ResourcesLayout>
+    <>
```
This is a temp banner for reviewing lesson quality will be removed before merged

JBR90 commented on apps/nextjs/src/components/AppComponents/AdditionalMaterials/Glossary.tsx:
```diff
@@ -19,6 +19,7 @@ export const Glossary = ({
         <OakFlex
           key={`${item.term}-${index}`}
           $flexDirection="row"
+          $justifyContent=""
```
🔥

codeincontext commented on packages/additional-materials/src/documents/additionalMaterials/promptHelpers.ts:
```diff
@@ -181,6 +181,11 @@ export const getLessonDetails = (
+    lines.push("**Transcript**:");
+    lines.push(lessonPlan.transcript);
+  }
+  console.log(lines);
```
🔥

codeincontext commented on packages/api/src/router/additionalMaterials.ts:
```diff
@@ -30,11 +22,224 @@ import {
+        userId: ctx.auth.userId,
+      });
+
+      const AUTH_KEY = process.env.CURRICULUM_API_AUTH_KEY;
+      const AUTH_TYPE = process.env.CURRICULUM_API_AUTH_TYPE;
+      const GRAPHQL_ENDPOINT = process.env.CURRICULUM_API_URL;
+
+      if (!AUTH_KEY || !AUTH_TYPE || !GRAPHQL_ENDPOINT) {
+        log.error("Missing environment variables", {
+          AUTH_KEY: !!AUTH_KEY,
+          AUTH_TYPE: !!AUTH_TYPE,
+          GRAPHQL_ENDPOINT: !!GRAPHQL_ENDPOINT,
+        });
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: "Missing required environment variables",
+        });
+      }
```
It could be worth extracting this to a function

JBR90 commented on packages/api/src/router/owaLesson/copyrightHelper.ts:
```diff
@@ -0,0 +1,43 @@
+const log = aiLogger("additional-materials");
+
+const RESTRICTED_CONTENT_GUIDANCE_TYPES = [
+  "Depiction or discussion of sexual violence",
+  "Depiction or discussion of sexual content",
+  "Depiction or discussion of mental health issues",
+  "Depiction or discussion of serious crime",
```
Yeah good point, hopefully the exist in the oak curriculum schema

JBR90 commented:
> I've added some notes in Notion and passed these to @JBR90 and @annac14 to review. I don't think any of them explicitly block this PR, the basic overall function seems to work. We could look at approving this one and then do further testing in OWA with the main URL link available from there.

I've fixed the mobile padding on start again and removed the 'x' on the error modal

mikeritson-oak reviewed (approved):
The work done in this PR looks good. 👍 
If we can get this merged ASAP, I can include this in the next release. Further testing can take place in [#3643](https://github.com/oaknational/Oak-Web-Application/pull/3643), once we have the work linked into OWA and more test data available.

oak-machine-user commented:
:tada: This PR is included in version 1.48.0 :tada:

The release is available on [GitHub release](https://github.com/oaknational/oak-ai-lesson-assistant/releases/tag/v1.48.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
